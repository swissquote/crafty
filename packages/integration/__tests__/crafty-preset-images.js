/* global jest, describe, it, expect */

const fs = require("fs");
const path = require("path");
const configuration = require("@swissquote/crafty/src/configuration");
const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

const getCrafty = configuration.getCrafty;

it("Loads crafty-preset-images and does not register gulp tasks", () => {
  const crafty = getCrafty(["@swissquote/crafty-preset-images"], {});

  const loadedPresets = crafty.config.loadedPresets.map(
    preset => preset.presetName
  );
  expect(loadedPresets).toContain("@swissquote/crafty-preset-images");

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([]);
});

it("Loads crafty-preset-images, crafty-runner-gulp and registers gulp task", () => {
  const crafty = getCrafty(
    ["@swissquote/crafty-preset-images", "@swissquote/crafty-runner-gulp"],
    {}
  );

  const loadedPresets = crafty.config.loadedPresets.map(
    preset => preset.presetName
  );
  expect(loadedPresets).toContain("@swissquote/crafty-preset-images");
  expect(loadedPresets).toContain("@swissquote/crafty-runner-gulp");

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([
    "images_all",
    "images_svg",
    "images",
    "default"
  ]);
});

it("Copies and compresses images", async () => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-images");

  const result = await testUtils.run(["run", "images"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/images/batman.svg")).toBeTruthy();
  expect(
    testUtils.exists(cwd, "dist/images/somedir/cute-cats-2.jpg")
  ).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/images/notcopied.txt")).toBeFalsy();

  expect(
    fs.statSync(path.join(cwd, "dist/images/batman.svg")).size
  ).toBeLessThan(fs.statSync(path.join(cwd, "images/batman.svg")).size);
  expect(
    fs.statSync(path.join(cwd, "dist/images/somedir/cute-cats-2.jpg")).size
  ).toBeLessThan(
    fs.statSync(path.join(cwd, "images/somedir/cute-cats-2.jpg")).size
  );
});
