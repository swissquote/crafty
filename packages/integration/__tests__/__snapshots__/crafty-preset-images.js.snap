// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copies and compresses images 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            img: __PATH__/packages/integration/fixtures/crafty-preset-images/dist/images
[__:__:__] Starting 'images' ...
[__:__:__] Starting 'images_all' ...
[__:__:__] gulp-imagemin: Minified 1 image (saved 518 kB - 76.7%)
[__:__:__] Finished 'images_all' after ____ ms
[__:__:__] Starting 'images_svg' ...
[__:__:__] Finished 'images_svg' after ____ ms
[__:__:__] Finished 'images' after ____ ms
",
}
`;
